src/Parallel/Invoke.hpp

```diff
@@ -17,7 +17,11 @@ namespace detail {
 inline bool max_inline_entry_methods_reached() noexcept {
   thread_local size_t approx_stack_depth = 0;
   approx_stack_depth++;
+#ifdef SPECTRE_PROFILING
+  if (true) {
+#else
   if (approx_stack_depth < 64) {
+#endif
```

Comment:
Hold up ... I'm confused by what's happened here.

I would expect that, for clarify of tracing, we're looking to _avoid_ inlining all the entry methods.

This means this function should always say "yes, we have reached the max inline entry methods, you can't inline anymore"... in other words the check `if (approx_stack_depth < 64)` should become `if (approx_stack_depth < 0)`, i.e., `if (false)`.

Is this a bug, or do I misunderstand?

In addition to this logic complaint, I suggest perhaps it might be easier to avoid the dummy `if` statement by writing this as,
```
inline bool max_inline_entry_methods_reached() noexcept {
#ifdef SPECTRE_PROFILING
  return true;  // or false if I misunderstand :P
#else
  // all the current code
#endif
```

Reply:
:facepalm: Yeah, that's my bad -- I reversed the logic - @fmahebert is right.